### ISSUE TITLE
Android: asking for microphone permission should (also) occure in singing scene

### ISSUE THREAD

**Issue by supernevi** [UltraStar-Deluxe/Play#301]
Issue type: Feature request

### Actual behaviour
After I just installed the android version, downloaded the demo music und wanted to start singing, the app just displays that there could not be found a mic.
This popes up only if I navigate to the microphone settings. After I granted access there still were no found microphones. And in addition no clicks were possible anymore. Only after restarting the app mics were found (maybe this should be another ticket)

**Comment by achimmihca** [comment]
Thanks for pointing this out.

> the app just displays that there could not be found a mic

This is on purpose.

You first have to enable a mic for singing. This requires users to enter the recording options.

Even on Android there can be multiple recording devices (my phone has "Android audio input" and "Android camcorder input" and something else). I don't want to make assumptions about devices, especially when it comes to desktop but also on mobile.

> Only after restarting the app mics were found

I also noticed this: must kill then restart app on Android for newly set permissions to become effective.
Such things make the whole app appear buggy as a first impression.

However, I also had this issue with other native Android apps and on different devices.

To be honest, I don't know what we could do about it. Maybe restart the recording options scene? But I am not even sure that Unity API provides a callback to react to changed permissions.
Unity provides only few methods to ask for permissions on Android.

Low prio because easy to work around (restart app).
But this should at least be documented as known issue in the wiki.
And would be great if someone takes a look into this (search Unity forums etc.)

> And in addition no clicks were possible anymore

This might actually be a bug on our side, or Unity having trouble with the newly set permissions.
But to think positively: at least it encourages the user to restart the app 😅

**Comment by achimmihca** [comment]
Unity docs: https://docs.unity3d.com/Manual/android-RequestingPermissions.html

USPlay somehow has to check if permissions have been set and if yes, probably needs to restart the recording options scene.

Or maybe restart the app after asking the user in a dialog.

**Comment by achimmihca** [comment]
Here are some helper methods for Android permissions:
https://github.com/yasirkula/UnityAndroidRuntimePermissions
